File: android/app/src/main/java/com/ethiobalance/app/data/SmsDao.java
package com.ethiobalance.app.data;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import java.util.List;

@Dao
public interface SmsDao {
    @Insert
    void insert(SmsEntity sms);

    @Query("SELECT * FROM sms_events WHERE isSynced = 0 ORDER BY timestamp ASC")
    List<SmsEntity> getUnsyncedSms();

    @Update
    void update(SmsEntity sms);

    @Query("UPDATE sms_events SET isSynced = 1 WHERE id IN (:ids)")
    void markAsSynced(List<Integer> ids);
}

File: android/app/src/main/java/com/ethiobalance/app/data/SmsDao.kt
package com.ethiobalance.app.data

import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Dao
interface SmsDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(sms: SmsEntity)

    @Query("SELECT * FROM sms_events WHERE isSynced = 0 ORDER BY timestamp ASC")
    suspend fun getUnsyncedSms(): List<SmsEntity>

    @Update
    suspend fun update(sms: SmsEntity)

    @Query("UPDATE sms_events SET isSynced = 1 WHERE id IN (:ids)")
    suspend fun markAsSynced(ids: List<Int>)

    @Query("SELECT * FROM sms_events ORDER BY timestamp DESC")
    fun getAllSmsFlow(): Flow<List<SmsEntity>>
}
